app/main.py

```diff
@@ -1,8 +1,50 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(
+            self, comfort_class: int,
+            clean_mark: int,
+            brand: str
+    ) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(
+            self, distance_from_city_center: float,
+            clean_power: int,
+            average_rating: float,
+            count_of_ratings: int
+    ) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list) -> float:
+        total_income = 0
+
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                income = self.calculate_washing_price(car)
+                total_income += income
+                self.wash_single_car(car)
+
+        return round(total_income, 1)
+
+    def calculate_washing_price(self, car: Car) -> float:
+        return (
+            (car.comfort_class * (self.clean_power - car.clean_mark)
+             * self.average_rating) / self.distance_from_city_center
+        )
+
+    def wash_single_car(self, car: Car) -> None:
+        if self.clean_power > car.clean_mark:
+            car.clean_mark = self.clean_power
```

Comment:
checking for the difference between “clean_power” and “clean_mark” seems unnecessary, since this is already checked in the “serve_cars” `method

Reply:
Not quite, it's better to leave it, because there may be a case when you need to reuse this method, and there will be no logic to check whether it is worth washing the car or not